crates/typed-store/src/rocks/iter.rs

```diff
@@ -58,10 +66,12 @@ impl<'a, K: DeserializeOwned, V: DeserializeOwned> Iterator for Iter<'a, K, V> {
                 .db_iter
                 .value()
                 .expect("Valid iterator failed to get value");
-            self.db_metrics
-                .rocksdb_iter_bytes
-                .with_label_values(&[&self.cf])
-                .observe((raw_key.len() + raw_value.len()) as f64);
+            if let Some(_timer) = report_metrics {
```

Comment:
Just a quick note here - this line of code did change what `_timer` of `rocksdb_iter_latency_seconds ` measures.

It previously measured time from start function till the end.

We this change it measure time from start of the function until end of this `if` block where _timer is dropped.

I am not sure how critical this is, but just pointing out that there is this subtle change.

Reply:
Ah, interesting.

Reply:
ah good catch, i'll move it down after deserialize

Reply:
You can also just use `report_metrics.is_some()` here